test/libdnf/private_accessor.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright Contributors to the libdnf project.
```

Comment:
I am actually not sure about this copyright notice, since the code is copy-pasted from someone else...

Reply:
Good point, because of movement of the project, the information is outdated. Anyway, for the PR I have no problem to keep it like i is.

Reply:
Hmm, I'm not sure you understand, this is unrelated to moving the project, I'm talking about the code in this file, which is copy-pasted from https://gist.github.com/dabrahams/1528856

I don't think it'll cause issues, but I thought I'd mention it.

Reply:
Thank you for additional information. Well in this case it can be a problem. Do you know the license under which the code was published?

Reply:
The linked gist doesn't have a license notice, so no. It is itself based on someone else's code: http://bloglitb.blogspot.com/2010/07/access-to-private-members-thats-easy.html (so asking one author may not be enough...)

Inferring from the way it's published I'd say it's public domain. I just thought I'd mention it for completeness...

Reply:
I tried to come up with my own version. I thought of several solutions. They all use the same idea. The address of private members can be obtained during explicit instantiation of the class.

```
// A getter generating structure
template<typename AccessTag, auto value>
struct CreateGetter {
    friend constexpr auto get(AccessTag) { return value; }
};

// Helper macro
#define create_getter(AccessTag, class_member_ptr) \
struct AccessTag {}; \
constexpr auto get(AccessTag); \
template struct CreateGetter<AccessTag, class_member_ptr>;
```

Example of usage:
```
// Class with private member
class A {
private:
    int priv_value = 10;
};

// Just use the macro to create the AccessTag and the getter
create_getter(A_priv_value, &A::priv_value)

// And use getter
int main() {
    A a;
    std::cout << a.*get(A_priv_value()) << std::endl; // read value
    a.*get(A_priv_value()) = 5;  // set value
    std::cout << a.*get(A_priv_value()) << std::endl; // read value
    return 0;
}
```

Reply:
Is it possible for tests import a private class from libdnf5/libdnf?